File: app/src/main/java/com/mmartin/ghibliapi/film/FilmsContract.java
package com.mmartin.ghibliapi.film;

import com.mmartin.ghibliapi.BasePresenter;
import com.mmartin.ghibliapi.BaseView;

/**
 * Created by mmartin on 9/12/17.
 */

public interface FilmsContract {
    interface View extends BaseView<Presenter> {
    }

    interface Presenter extends BasePresenter {
    }
}

File: app/src/main/java/com/mmartin/ghibliapi/film/FilmsContract.kt
package com.mmartin.ghibliapi.film

import com.mmartin.ghibliapi.BasePresenter
import com.mmartin.ghibliapi.BaseView
import com.mmartin.ghibliapi.data.model.Film

interface FilmsContract {
    interface View : BaseView {
        fun showFilms(filmList: List<Film>)
        fun showError()
    }

    interface Presenter : BasePresenter<View> {
        fun loadFilms()
    }
}